test/src/unit-QueryCondition.cc

```diff
@@ -2454,6 +3073,100 @@ TEST_CASE(
+
+  // Initialize the array schema.
+  ArraySchema array_schema;
+  Attribute attr(field_name, type);
+  REQUIRE(array_schema.add_attribute(tdb::make_shared<Attribute>(HERE(), &attr))
+              .ok());
+  Domain domain;
+  Dimension dim("dim1", Datatype::UINT32);
+  uint32_t bounds[2] = {1, cells};
+  Range range(bounds, 2 * sizeof(uint32_t));
+  REQUIRE(dim.set_domain(range).ok());
+  REQUIRE(
+      domain
+          .add_dimension(tdb::make_shared<tiledb::sm::Dimension>(HERE(), &dim))
+          .ok());
+  REQUIRE(
+      array_schema.set_domain(make_shared<tiledb::sm::Domain>(HERE(), &domain))
+          .ok());
+
+  // Initialize the result tile.
+  ResultTile result_tile(0, 0, array_schema);
+  result_tile.init_attr_tile(field_name);
+  ResultTile::TileTuple* const tile_tuple = result_tile.tile_tuple(field_name);
+  Tile* const tile = &std::get<0>(*tile_tuple);
+
+  // Initialize and populate the data tile.
+  REQUIRE(tile->init_unfiltered(
+                  constants::format_version,
+                  type,
+                  cells * sizeof(uint64_t),
+                  sizeof(uint64_t),
+                  0)
+              .ok());
+  uint64_t* values = static_cast<uint64_t*>(malloc(sizeof(uint64_t) * cells));
+  for (uint64_t i = 0; i < cells; ++i) {
+    values[i] = i;
+  }
+  REQUIRE(tile->write(values, 0, cells * sizeof(uint64_t)).ok());
+
+  // Build a combined query for `> 6 AND <= 3`.
+  uint64_t cmp_value_1 = 6;
+  QueryCondition query_condition_1;
+  REQUIRE(query_condition_1
+              .init(
+                  std::string(field_name),
+                  &cmp_value_1,
+                  sizeof(uint64_t),
+                  QueryConditionOp::GT)
+              .ok());
+  // Run Check
+  REQUIRE(query_condition_1.check(array_schema).ok());
+  uint64_t cmp_value_2 = 3;
+  QueryCondition query_condition_2;
+  REQUIRE(query_condition_2
+              .init(
+                  std::string(field_name),
+                  &cmp_value_2,
+                  sizeof(uint64_t),
+                  QueryConditionOp::LE)
+              .ok());
+  // Run Check
+  REQUIRE(query_condition_2.check(array_schema).ok());
+  QueryCondition query_condition_3;
+  REQUIRE(query_condition_1
+              .combine(
+                  query_condition_2,
+                  QueryConditionCombinationOp::OR,
+                  &query_condition_3)
+              .ok());
+
+  // Apply the query condition.
+  uint64_t cell_count = 0;
+  std::vector<uint8_t> result_bitmap(cells, 1);
+  REQUIRE(query_condition_3
+              .apply_sparse<uint8_t>(
+                  array_schema, result_tile, result_bitmap, &cell_count)
+              .ok());
+
+  // Check that the cell slab now contains cell indexes 0, 1, 2, 3, and 7, 8, 9.
+  REQUIRE(cell_count == 7);
+  for (uint64_t cell_idx = 0; cell_idx < cells; ++cell_idx) {
+    REQUIRE(
+        result_bitmap[cell_idx] == (cell_idx >= 7 || cell_idx <= 3 ? 1 : 0));
+  }
+
+  free(values);
+}
+
```

Comment:
Should probably add a test combining or with and - two different ways -- an or with an and plus an and with an or.